### PR TITLE
Introduce ReadBlockOrHeaderFromDisk

### PR DESCRIPTION
**Pull request by rnicoll**

### CODE REVIEW

File: src/validation.cpp [hunk 1]
```diff
@@ -1161,7 +1166,7 @@ bool ReadBlockFromDisk(CBlock& block, const FlatFilePos& pos, const Consensus::P
     }
 
     // Check the header
-    if (!CheckProofOfWork(block.GetPoWHash(), block.nBits, consensusParams))
+    if (fCheckPOW && !CheckProofOfWork(block.GetPoWHash(), block.nBits, consensusParams))
         return error("ReadBlockFromDisk: Errors in block header at %s", pos.ToString());
```
Comment (patricklodder) [inline src/validation.cpp:1170]: This says "ReadBlockOrHeader:" currently in 1.14, which seems better than referring to the old function name.

**Review by patricklodder — CHANGES_REQUESTED** [review]

File: src/chain.cpp [hunk 2]
```diff
@@ -5,6 +5,10 @@
 
 #include <chain.h>
 
+// Dogecoin: Include the single function signature we need, to avoid the entire
+// validation.h being pulled in and creating a cyclic include loop
+bool ReadBlockHeaderFromDisk(CBlockHeader& block, const CBlockIndex* pindex, const Consensus::Params& consensusParams, const bool fCheckPOW = true);
+
```
[the remark this answers is not in the document]
Comment (rnicoll) [reply src/chain.cpp]: Done

File: src/validation.cpp (hunk 1 above)
Comment (rnicoll) [reply src/validation.cpp]: Done

**Review by patricklodder — APPROVED** [review]
Checked compile on 20.04 - works. Code is clean.

utACK

**Pull request merged**